libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -296,8 +296,7 @@ struct AdaptiveOrderedByIntegerMetric : private boost::noncopyable {
         ((double)(p.slowPopsLastPeriod) / (double)(p.sinceLastFix)) >
             1.0 / (double)(chunk_size)) {
       for (unsigned i = 1; i < runtime::activeThreads; ++i) {
-        while (current.getRemote(i)->lock.try_lock())
```

Comment:
Here it is the culprit - something is missing`!`

Reply:
Great find. That totally explains why it was broken before. Thanks!